src/lando/main/tests/test_git.py

```diff
@@ -503,3 +503,91 @@ def test_GitSCM_merge_onto(
         assert (
             merged_file == expected_content
         ), f"File contents did not match expected for strategy {strategy}"
+
+
+def test_GitSCM_tag(
+    git_repo: Path,
+    git_setup_user: Callable,
+    request: pytest.FixtureRequest,
+    tmp_path: Path,
+):
+    clone_path = tmp_path / request.node.name
+    clone_path.mkdir()
+
+    scm = GitSCM(str(clone_path))
+    scm.clone(str(git_repo))
+    git_setup_user(str(clone_path))
+
+    # Create a new commit and get its SHA
+    _create_git_commit(request, clone_path)
+    commit_sha = (
+        subprocess.run(
+            ["git", "rev-parse", "HEAD"],
+            cwd=clone_path,
+            capture_output=True,
+            check=True,
+        )
+        .stdout.decode()
+        .strip()
+    )
+
+    # Tag the current commit
+    tag_name = "v1.0"
+    scm.tag(tag_name, None)
+
+    # Check that the tag exists
+    tag_output = (
+        subprocess.run(
+            ["git", "tag", "--list"], cwd=clone_path, capture_output=True, check=True
+        )
+        .stdout.decode()
+        .splitlines()
+    )
+
+    assert tag_name in tag_output, f"New tag {tag_name} should be present in tags list."
+
+    # Check the tag points to the expected commit
+    tag_sha = (
+        subprocess.run(
+            ["git", "rev-list", "-n", "1", tag_name],
+            cwd=clone_path,
+            capture_output=True,
+            check=True,
+        )
+        .stdout.decode()
+        .strip()
+    )
+
+    assert tag_sha == commit_sha, "Tag should point to expected commit."
+
+
+def test_GitSCM_push_tag(
+    git_repo: Path,
+    git_setup_user: Callable,
+    request: pytest.FixtureRequest,
+    tmp_path: Path,
+):
+    clone_path = tmp_path / request.node.name
+    clone_path.mkdir()
+
+    scm = GitSCM(str(clone_path))
+    scm.clone(str(git_repo))
+    git_setup_user(str(clone_path))
```

Comment:
This comment applies to other similar tests in this module.

These lines should be in a fixture (repeated above).
Side note, why not use `repo_mc` to keep things consistent/abstracted with the push/pull paths and other tests? Slightly more high level but still worth it if it simplifies these tests I think.

Reply:
I used the existing conventions in the other tests in this file, which don't use the `repo_mc` fixture. I agree that we should try and keep things more consistent, but let's do it as a follow-up since fixing all the tests in this file is out of scope.